x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/transport/DualStackCoordinator.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.security.transport;
+
+import org.elasticsearch.common.network.CloseableChannel;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.util.concurrent.ConcurrentCollections;
+import org.elasticsearch.xpack.core.XPackSettings;
+
+import java.util.Set;
+
+public class DualStackCoordinator {
+
+    private volatile Set<CloseableChannel> plaintextChannels = null;
+
+    public DualStackCoordinator(ClusterSettings clusterSettings) {
```

Comment:
I don't think we should make this a dynamic setting. instead I think we should make it a node level setting that disables itself once all nodes are on SSL? I think this logic is too complicated, we should just do this based on what's in the node-settings. Once all connections are SSL we should disable it automatically. Thsi should be save since if the cluster is already upgraded the master will be SSL only and therefore we can trust it?